Waku post

oskarth opened the pull request:
Initial draft

decanus commented on _posts/2020-02-10-waku-current-state.md:
```diff
@@ -0,0 +1,141 @@
+
+# Where we are at with Waku?
+
+Waku is our fork of Whisper where we address the shortcomings of Whisper in an iterative manner. We've seen in previous post that Whisper doesn't scale, and why. Here's an update on where we are at with Waku since that [last post](https://vac.dev/fixing-whisper-with-waku).
```
```suggestion
Waku is our fork of Whisper where we address the shortcomings of Whisper in an iterative manner. We've seen a in [previous post](https://vac.dev/fixing-whisper-with-waku) that Whisper doesn't scale, and why. Here's an update on where we are at with Waku since then.
```

decanus commented on _posts/2020-02-10-waku-current-state.md:
```diff
@@ -0,0 +1,141 @@
+
+There are currently two clients that implement Waku, these are [Nimbus](https://github.com/status-im/nimbus/tree/master/waku) in Nim and [status-go](https://github.com/status-im/status-go) in Go.
+
+At the time of this writing the Nimbus client implements the spec fully, but lacks mail server and mail client capability. The status-go client implements everything except bridging mode, which is currently a work in progress.
```
```suggestion
At the time of writing this, the Nimbus client implements the spec fully, but lacks mail server and mail client capability. The status-go client implements everything except bridging mode, which is currently a work in progress.
```

oskarth replied:
```suggestion
At the time of writing the Nimbus client implements the spec fully, but lacks mail server and mail client capability. The status-go client implements everything except bridging mode, which is currently a work in progress.
```
https://english.stackexchange.com/questions/199381/is-at-the-time-of-writing-correct/199392

decanus commented on _posts/2020-02-10-waku-current-state.md:
```diff
@@ -0,0 +1,141 @@
+
+**Simulation:**
+
+We've got a [simulation](https://github.com/status-im/nimbus/tree/master/waku#testing-waku-protocol) in the Nimbus client that verifies - or rather, fails to falsify - the scalability model posted in an [earlier post](https://vac.dev/fixing-whisper-with-waku). More on this below.
```
```suggestion
We've got a [simulation](https://github.com/status-im/nimbus/tree/master/waku#testing-waku-protocol) in the Nimbus client that verifies - or rather, fails to falsify - the scalability model described in an [earlier post](https://vac.dev/fixing-whisper-with-waku). More on this below.
```

decanus commented on _posts/2020-02-10-waku-current-state.md:
```diff
@@ -0,0 +1,141 @@
+
+## Difference between Waku and Whisper
+
+Summary of main differences between Waku v0 spec and Whisper v6, as described in EIP-627:
```
```suggestion
Summary of main differences between Waku v0 spec and Whisper v6, as described in [EIP-627](https://eips.ethereum.org/EIPS/eip-627):
```

kdeme commented on _posts/2020-02-10-waku-current-state.md:
```diff
@@ -0,0 +1,141 @@
+
+Here we outline two network topologies, Star and full mesh, with 6 randomly connected nodes and 2 X, sending 10000 random envelopes over random topics, including invalid envelopes.
+
+For light node, bloom filter is set to 1% false positive. It shows the number of valid and invalid envelopes received for the different nodes.
```
With the settings in the quicksim the false positive probability is almost 10% (p=0.087). 
Light node and Waku light node subscribe to 100 randomly selected topics (that makes n=100, k=3, m=512 for the bloom filter). 

Next, one of the full nodes sends 1 envelope over 1 of the 100 topics that the two light nodes subscribed on. This is to test that the subscription works, else the 10k envelopes will of course not be received (come to think of it, this should be done for all 100 topics to be a proper check).
Then, that same full node sends 10000 envelopes over random topics.

This explains why you see 10001 for the full nodes, and 1 for the Waku light node (as in fact, with just 10000 envelopes to random topics, the chance is very low that one of the Waku light node topics gets "hit").

oskarth replied:
Ah should've looked more carefully, thanks for the correction and added detail :)

oskarth replied:
fixed

kdeme commented on _posts/2020-02-10-waku-current-state.md:
```diff
@@ -0,0 +1,141 @@
+
+There are currently two clients that implement Waku, these are [Nimbus](https://github.com/status-im/nimbus/tree/master/waku) in Nim and [status-go](https://github.com/status-im/status-go) in Go.
+
+At the time of this writing the Nimbus client implements the spec fully, but lacks mail server and mail client capability. The status-go client implements everything except bridging mode, which is currently a work in progress.
```
Not sure if this detail needs to be added, but the nim implementation also lacks rate limiting and confirmations. But these are optional so in theory it still complies to spec.

oskarth replied:
added

decanus commented on _posts/2020-02-10-waku-current-state.md:
```diff
@@ -0,0 +1,141 @@
+
+This is our current understanding of how many users a network running the Waku protocol can support. Specifically in the context of the Status chat app, since that's the most immediate consumer of Waku. It should generalize fairly well to most deployments, but YMMW.
+
+**tldr (for Status app):**
```
```suggestion
**tl;dr (for Status app):**
```

decanus commented on _posts/2020-02-10-waku-current-state.md:
```diff
@@ -0,0 +1,141 @@
+
+*Assuming 10 concurrent users = 100 DAU. Estimate uncertainty increases for each order of magnitude until real-world data is observed.*
+
+As far as we know right now, there is one immediate bottleneck, a very likely bottleneck after that, and a third bottleneck that is still conjecture but not unlikely to appear. These are:
```
```suggestion
As far as we know right now, these are the bottlenecks we have:
```

oskarth replied:
Hm, there's only one bottleneck at a time though? They don't appear all at once

oskarth replied:
I suppose you edit this below

decanus commented on _posts/2020-02-10-waku-current-state.md:
```diff
@@ -0,0 +1,141 @@
+
+As far as we know right now, there is one immediate bottleneck, a very likely bottleneck after that, and a third bottleneck that is still conjecture but not unlikely to appear. These are:
+
+- Bottleneck 1 - Receive bandwidth for end user clients (aka ‘Fixing Whisper with Waku’)
```
```suggestion
- Immediate bottleneck - Receive bandwidth for end user clients (aka ‘Fixing Whisper with Waku’)
```

decanus commented on _posts/2020-02-10-waku-current-state.md:
```diff
@@ -0,0 +1,141 @@
+As far as we know right now, there is one immediate bottleneck, a very likely bottleneck after that, and a third bottleneck that is still conjecture but not unlikely to appear. These are:
+
+- Bottleneck 1 - Receive bandwidth for end user clients (aka ‘Fixing Whisper with Waku’)
+- Bottleneck 2 - Nodes and cluster capacity (aka ‘DNS based node discovery’)
```
```suggestion
- Very likely bottleneck - Nodes and cluster capacity (aka ‘DNS based node discovery’)
```

decanus commented on _posts/2020-02-10-waku-current-state.md:
```diff
@@ -0,0 +1,141 @@
+
+- Bottleneck 1 - Receive bandwidth for end user clients (aka ‘Fixing Whisper with Waku’)
+- Bottleneck 2 - Nodes and cluster capacity (aka ‘DNS based node discovery’)
+- Bottleneck 3 - Full node traffic (aka ‘the routing / partition problem’)
```
```suggestion
- Conjecture but not unlikely to appear- Full node traffic (aka ‘the routing / partition problem’)
```

oskarth commented on _posts/2020-02-10-waku-current-state.md:
```diff
@@ -0,0 +1,142 @@
+
+There are currently two clients that implement Waku, these are [Nimbus](https://github.com/status-im/nimbus/tree/master/waku) in Nim and [status-go](https://github.com/status-im/status-go) in Go.
+
+At the time of writing the Nimbus client implements the spec fully, but lacks mail server/client, rate limiting and confirmations capability. The status-go client implements everything except bridging mode, which is currently a work in progress.
```
@adambabik please confirm this is accurate, ideally with link to whats left

adambabik commented on _posts/2020-02-14-where-are-we-at-with-waku.md:
```diff
@@ -0,0 +1,147 @@
+
+There are currently two clients that implement Waku, these are [Nimbus](https://github.com/status-im/nimbus/tree/master/waku) in Nim and [status-go](https://github.com/status-im/status-go) in Go.
+
+At the time of writing the Nimbus client implements the spec fully, but lacks mail server/client, rate limiting and confirmations capability. The status-go client implements everything except bridging mode, which is currently a work in progress.
```
status-go has partial rate-limiting implementation. There is still work to be done yet in terms of broadcasting it and enforcing.

adambabik commented on _posts/2020-02-14-where-are-we-at-with-waku.md:
```diff
@@ -0,0 +1,147 @@
+- Handshake/Status message not compatible with shh/6 nodes; specifying options as association list
+- Include topic-interest in Status handshake
+- Upgradability policy
+- `topic-interest` packet code
```
There has been some developments around it to remove it and replace with a packet to exchange `statusOptions` like in the handshake.

decanus commented on _posts/2020-02-14-where-are-we-at-with-waku.md:
```diff
@@ -0,0 +1,147 @@
+
+For more details, see [implementation matrix](https://specs.vac.dev/waku/waku.html#appendix-b-implementation-notes).
+
+In terms of end user applications, work is currently in progress to integrate it into the [Status core app](https://github.com/status-im/status-react/pull/9949) and is expected to be released in their upcoming 1.1 release.
```
any link to show its coming in 1.1?

oskarth replied:
added link to roadmap, no better link right now

decanus commented on _posts/2020-02-14-where-are-we-at-with-waku.md:
```diff
@@ -0,0 +1,147 @@
+
+## How many users does Waku support?
+
+This is our current understanding of how many users a network running the Waku protocol can support. Specifically in the context of the Status chat app, since that's the most immediate consumer of Waku. It should generalize fairly well to most deployments, but YMMW.
```
what is YMMW?

oskarth replied:
your milage might vary (typo)

removed

oskarth commented:
Addressed comments. Removing you as reviewer @decanus as I fixed those things and want to push it live to preview. We can always amend in follow up PR
